unified-runtime/source/adapters/cuda/context.hpp

```diff
@@ -190,6 +170,59 @@ struct ur_context_handle_t_ {
   std::mutex Mutex;
   std::vector<deleter_data> ExtendedDeleters;
   std::set<ur_usm_pool_handle_t> PoolHandles;
```

Comment:
Not related to this PR but, could you add a comment describing what are `PoolHandles` are? It would make understanding the code a bit easier.

Reply:
`A set of user-created USM Disjoint Pools` ? - is it OK?